### ISSUE TITLE
Error when compiling on OSX 10.11 with clang-3.8svn 

### ISSUE THREAD

**Issue by jcar87** [PointCloudLibrary/pcl#1458]
Compiler: clang-3.8 (from llvm svn. This one supports OpenMP as opposed to Apple's one)
OSX : 10.11.1 El Capitan
Eigen version: 3.2.0

Build error happens with both 1.7.2 versions and up-to-date git
```
[ 35%] Building CXX object features/CMakeFiles/pcl_features.dir/src/narf.cpp.o
/Users/jcc/dev/pcl-git/features/src/narf.cpp:383:22: error: no viable
      conversion from 'Map<Matrix<[3 * ...]>>' to 'const Map<const Matrix<[3
      * ...]>>'
  ...point = interest_points.points[interest_point_idx].getVector3fMap ();
     ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/eigen3/Eigen/src/Core/Map.h:104:79: note: candidate
      constructor (the implicit copy constructor) not viable: no known
      conversion from 'pcl::Vector3fMap' (aka 'Map<Matrix<float, 3, 1> >') to
      'const Eigen::Map<const Eigen::Matrix<float, 3, 1, 0, 3, 1>, 0,
      Eigen::Stride<0, 0> > &' for 1st argument
  ...PlainObjectType, int MapOptions, typename StrideType> class Map
                                                                 ^
/usr/local/include/eigen3/Eigen/src/Core/Map.h:139:12: note: candidate
      constructor not viable: no known conversion from 'pcl::Vector3fMap' (aka
      'Map<Matrix<float, 3, 1> >') to 'PointerArgType' (aka 'const float *') for
      1st argument
    inline Map(PointerArgType dataPtr, const StrideType& a_stride = Stri...
           ^
1 error generated.
make[2]: *** [features/CMakeFiles/pcl_features.dir/src/narf.cpp.o] Error 1
make[1]: *** [features/CMakeFiles/pcl_features.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by SergioRAgostinho** [comment]
Useful sidenote, [clang-omp](https://clang-omp.github.io/) is working as intended, and its included in homebrew. It could be a good workaround in the meantime. 
```
brew install clang-omp
```

If you want to contribute with a PR it should be enough, to replace
```cpp
point = interest_points.points[interest_point_idx].getVector3fMap ()`
```
with
```cpp
point = interest_points.points[interest_point_idx].getVector3fMapConst ()`
```
I tried to reproduce this compilation error in llvm-clang 3.6 (the one in homebrew) but its still not throwing these errors in that version.

**Comment by jcar87** [comment]
I guess I can give clang-omp a go. I went with a recent one because as far as I know clang-omp hasn't been updated for a while and won't be anymore.

I've tried compiling with:
clang-3.8svn (current svn revision), same error
clang-3.7 (official release), same error
Apple's clang (bundled with Xcode 7) - builds fine. This version of clang was built from clang-3.7svn. But no OpenMP support on this one

Replacing that line unfortunately failed:
`/features/src/narf.cpp:383:73: error: no member named 'getVector3fMapConst' in 'pcl::InterestPoint'
    Vector3fMapConst point = interest_points.points[interest_point_idx].getVector3fMapConst ();`
